### PR TITLE
Add comparison operator for Direction and ElementId

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This is useful for optimizations of the elliptic code that I have on a branch right now. Also probably for other things.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/Structure/Direction.hpp [hunk 1]
```diff
@@ -181,6 +181,26 @@ bool operator!=(const Direction<VolumeDim>& lhs,
   return not(lhs == rhs);
 }
 
+template <size_t VolumeDim>
+bool operator<(const Direction<VolumeDim>& lhs,
+               const Direction<VolumeDim>& rhs);
+
+template <size_t VolumeDim>
+inline bool operator>(const Direction<VolumeDim>& lhs,
```
Comment (nilsdeppe) [inline src/Domain/Structure/Direction.hpp]: templates don't need to be marked as `inline`, they have the linkage already

File: src/Domain/Structure/Direction.hpp [hunk 2]
```diff
@@ -181,6 +181,26 @@ bool operator!=(const Direction<VolumeDim>& lhs,
   return not(lhs == rhs);
 }
 
+template <size_t VolumeDim>
+bool operator<(const Direction<VolumeDim>& lhs,
```
Comment (nilsdeppe) [inline src/Domain/Structure/Direction.hpp:187]: Maybe document (either dox or C++) what the comparison is and that it's arbitrary without some deep motivation behind it? I worry that in 2 years we'll look at this and ponder why it's comparing in this specific way :)

File: src/Domain/Structure/ElementId.hpp [hunk 3]
```diff
@@ -154,6 +154,26 @@ template <size_t VolumeDim>
 bool operator!=(const ElementId<VolumeDim>& lhs,
                 const ElementId<VolumeDim>& rhs);
 
+template <size_t VolumeDim>
+bool operator<(const ElementId<VolumeDim>& lhs,
```
Comment (nilsdeppe) [inline src/Domain/Structure/ElementId.hpp:165]: Maybe document (either dox or C++) what the comparison is and that it's arbitrary without some deep motivation behind it? I worry that in 2 years we'll look at this and ponder why it's comparing in this specific way :)

File: src/Domain/Structure/ElementId.hpp [hunk 4]
```diff
@@ -154,6 +154,26 @@ template <size_t VolumeDim>
...
+               const ElementId<VolumeDim>& rhs);
+
+template <size_t VolumeDim>
+inline bool operator>(const ElementId<VolumeDim>& lhs,
```
Comment (nilsdeppe) [inline src/Domain/Structure/ElementId.hpp]: don't need `inline` on templates (below too)

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Overall LGTM, couple minor things. Please squash and rebase immediately

**Comment by nilsvu** [comment]
Done, thanks @nilsdeppe!

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**